cirq-core/cirq/optimizers/two_qubit_to_sqrt_iswap.py

```diff
@@ -0,0 +1,340 @@
+"""Utility methods for decomposing two-qubit unitaries into sqrt-iSWAP gates.
+
+References:
+    Towards ultra-high fidelity quantum operations: SQiSW gate as a native
+    two-qubit gate
+    https://arxiv.org/abs/2105.06074
+"""
+
+from typing import Optional, Sequence, Tuple, TYPE_CHECKING
+
+import numpy as np
+
+from cirq import ops, linalg, protocols
+from cirq.optimizers import two_qubit_decompositions
+
+if TYPE_CHECKING:
+    import cirq
+
+
+def two_qubit_matrix_to_sqrt_iswap_operations(
+    q0: 'cirq.Qid',
+    q1: 'cirq.Qid',
+    mat: np.ndarray,
+    *,
+    required_sqrt_iswap_count: Optional[int] = None,
+    atol: float = 1e-8,
+    check_preconditions: bool = True,
+    clean_operations: bool = True,
+) -> Sequence['cirq.Operation']:
+    """Decomposes a two-qubit operation into Z/XY/sqrt-iSWAP gates.
+
+    Args:
+        q0: The first qubit being operated on.
+        q1: The other qubit being operated on.
+        mat: Defines the operation to apply to the pair of qubits.
+        required_sqrt_iswap_count: When specified, exactly this many sqrt-iSWAP
+            gates will be used even if fewer is possible (maximum 3).
+        atol: A limit on the amount of absolute error introduced by the
+            construction.
+        clean_operations: Enables optimizing resulting operation list by
+            merging operations and ejecting phased Paulis and Z operations.
+
+    Returns:
+        A list of operations implementing the matrix including at most three
+        SQRT_ISWAP (sqrt-iSWAP) gates, single-qubit gates, and a global phase
+        gate.
+
+    References:
+        Towards ultra-high fidelity quantum operations: SQiSW gate as a native
+        two-qubit gate
+        https://arxiv.org/abs/2105.06074
+    """
+    kak = linalg.kak_decomposition(
+        mat, atol=atol / 10, rtol=0, check_preconditions=check_preconditions
+    )
+    operations = _kak_decomposition_to_sqrt_iswap_operations(
+        q0, q1, kak, required_sqrt_iswap_count, include_global_phase=not clean_operations, atol=atol
+    )
+    if clean_operations:
+        return two_qubit_decompositions._cleanup_operations(operations)
+    return operations
+
+
+def _kak_decomposition_to_sqrt_iswap_operations(
+    q0: 'cirq.Qid',
+    q1: 'cirq.Qid',
+    kak: linalg.KakDecomposition,
+    required_sqrt_iswap_count: Optional[int] = None,
+    include_global_phase: bool = False,
+    atol: float = 1e-8,
+) -> Sequence['cirq.Operation']:
+    """Computes the list of operations in the sqrt-iSWAP decomposition."""
```

Comment:
This is not a particularly useful docstring.  Either expand it or delete it, since it just rephrases the function name.

Reply:
Removed.